src/sysext/sysext.c

```diff
@@ -61,6 +70,7 @@ static bool arg_no_reload = false;
 static int arg_noexec = -1;
 static ImagePolicy *arg_image_policy = NULL;
 static bool arg_varlink = false;
+static MutableMode arg_mutable = MUTABLE_AUTO;
```

Comment:
The default must be off, as this defeats the whole point of sysexts, so it needs to be opt-in

Reply:
This is opt-in on the filesystem level. Adding another command line flag requirement just for extra inconvenience is not good because it means one would need to overwrite the whole `ExecStart=` line of the unit file with a drop-in.

Reply:
We'll follow through on the request @bluca but please take note that your ask is a direct contradiction of what was requested one month ago in https://github.com/systemd/systemd/pull/31000#issuecomment-1903940042 (and duly implemented):

> i'd really like to see some switch btw, i.e. --mutable= or so (or --writable=), which parses as a boolean, but also takes additional values. i.e. if "yes" you get the writable thing, and we'll create the dir if missing. if "no" you will not get the writable thing at all, we'll ignore the writable dir. **if "auto" we'll use it if it exists but not if not (_which would be the default_)** and maybe as cherry on top "import" or so, which will use the writable layer, but only as yet another immutable layer, but not allow changes.

We sincerely hope that implementing your request will settle this issue.

Reply:
How about adding a config file that defines the desired behavior? The default for the config file could be "not mutable" but distros and users could overwrite it easily without having to change the units ExecStart.

Reply:
There's no need for an extra config file, the command line switch is more than enough